openlibrary/utils/open_syllabus_project.py

```
@@ -0,0 +1,108 @@
+    return None
+
+
+def generate_osp_db(input_directory: str) -> None:
+    """
+    This function generates an SQLite database from a directory of .json.gz files.
+    The database contains data extracted from the JSON files, including the OLID and total fields.
+    The function excludes lines where the 'total' is less than one.
+    The function creates an index on the OLID column for faster querying.
+
+    Args:
+        input_directory (str): The directory containing the .json.gz files.
+
+    Returns:
+        None
+    """
+
+    # Initialize a list to store the data
+    data = []
+
+    # Create an SQLite database and table
+    conn = sqlite3.connect(db_file)
+    cursor = conn.cursor()
+    # Drop the table if it exists so we only have fresh data
+    cursor.execute('DROP TABLE IF EXISTS data;')
+    cursor.execute(
+        '''
+        CREATE TABLE IF NOT EXISTS data (
+            olid INTEGER PRIMARY KEY,
+            total INTEGER
+        )
+    '''
+    )
+
+    i = 0
+    # Iterate through the files in the input directory
+    for filename in os.listdir(input_directory):
+        print(i)
+        i += 1
+        if filename.endswith(".json.gz"):
+            with gzip.open(os.path.join(input_directory, filename), "rt") as file:
+                for line in file:
+                    try:
+                        # Parse the JSON data
+                        json_data = json.loads(line)
+
+                        # Extract the 'ol_id' and 'total' fields
+                        ol_id = int(
+                            json_data["ol_id"].replace("/works/OL", "").replace("W", "")
+                        )
+                        total = json_data["total"]
+
+                        # Exclude lines where the 'total' is less than one
+                        if total >= 1:
+                            data.append((ol_id, total))
+                    except Exception as e:
+                        print(f"Error processing {filename}: {e}")
```

Since we don't know what the potential errors are ; I'd rather have the script crash . That would at least tell us. Otherwise we might have a systematic silent error that makes our data incorrect.